docs/source/integrations/mlflow.md

```diff
@@ -0,0 +1,278 @@
+to log the `run_params` passed to the Hook.
+An implementation would look as follows:
+
+```python
+import typing as t
+
+import mlflow
+from kedro.framework.hooks import hook_impl
+from mlflow.entities import RunStatus
+
+
+class MLflowHooks:
+    @hook_impl
+    def before_pipeline_run(self, run_params: dict[str, t.Any]):
+        logger.info("Logging run parameters to MLflow")
+        mlflow.log_params(run_params)
+        mlflow.set_tags({"pipeline": run_params["pipeline_name"] or "__default__"})
+
+    @hook_impl
+    def on_pipeline_error(self):
+        logger.error("Logging error to MLflow")
+        mlflow.end_run(RunStatus.to_string(RunStatus.FAILED))
+```
+
+Notice that there is also extra code to mark the run as failed if there is any error.
+
+After enabling this custom Hook, you can execute `kedro run`, and see something like this in the logs:
+
+```
+[05/06/24 12:43:30] INFO     Kedro project spaceflights-mlflow             session.py:324
+[05/06/24 12:43:31] INFO     Logging run parameters to MLFlow                 hooks.py:18
+...
+```
+
+If you open your tracking server UI you will observe a result like this:
+
+```{image} ../meta/images/simple-mlflow-tracking1.png
+:alt: Simple MLflow tracking
+:width: 50%
+:align: center
+```
+
+```{image} ../meta/images/simple-mlflow-tracking2.png
+:alt: Simple MLflow tracking
+:width: 50%
+:align: center
+```
+
+Notice that these "parameters" do not include the Kedro parameters as defined in `parameters*.yml`.
+To log the actual Kedro parameters, you can continue writing a custom Hook
+or use the `kedro-mlflow` plugin instead, as described in the following section.
+
+### Complete tracking of Kedro runs in MLflow using `kedro-mlflow`
+
+You could make the code snippet above more complex, but you are likely to reach a point where it is easier to take advantage of the work that the `kedro-mlflow` plugin can do for you.
+
+To start using `kedro-mlflow`, the only required step is to install it:
+
+```
+pip install kedro-mlflow
+```
+
+In recent versions of Kedro, this will already register the `kedro-mlflow` Hooks for you.
+
+:::{note}
+If you implemented your custom `MLflowHooks` as described in the previous section,
+consider disabling them to avoid logging the runs twice.
+:::
+
+`kedro-mlflow` will use a subset of the `run_params` as tags for the MLflow run,
+and log the Kedro parameters as MLflow parameters:
+
+```{image} ../meta/images/complete-mlflow-tracking-kedro-mlflow.png
+:alt: Complete MLflow tracking with kedro-mlflow
+:width: 50%
+:align: center
+```
+
+Check out {external+kedro-mlflow:doc}`the official kedro-mlflow tutorial <source/03_getting_started/02_first_steps>`
+for more detailed steps.
+
+Notice that `kedro-mlflow` assumes 1 Kedro project = 1 MLflow experiment.
+If this does not suit your use case, consider pursuing a more customised approach
+like the one described in the previous section.
```

Comment:
Actually this is the default but it is fully customisable in the ``mlflow.yml``, so I won't call it an assumption (you can even use a custom resolver to create something very custom and dynamic in the mlflow.yml if you want)

Reply:
Very true. I'll reword this section.

Reply:
Addressing in the upcoming version.